25_exceptions_lab/01.py

```diff
@@ -1,7 +1,31 @@
-"""
-Write a python program that takes numbers in a loop
-and for each number prints its square root
-If value is negative or not a number show 
-a warning and keep reading values
-"""
-
+'''
+Gets numbers as input from user and prints sqrt(number)
+If input is not a number or is a negative number, uses exceptions to show error
+'''
+
+import math
+
+
+def try_parse_float(string):
+    try:
+        return float(string)
+    except ValueError:
+        raise ValueError('Input \'%s\' is not a number' % string)
+
+
+def sqrt(n):
+    if type(n) == str:
+        n = try_parse_float(n)
+    if n < 0:
+        raise ValueError('Expected positive number, got %f' % n)
+    return math.sqrt(n)

```

Comment:
Hi Elad,
No need to add complexity where it's not required - 

Better (more pythonic) approach is to use python's built-in math.sqrt function directly, because it already raises the correct exceptions. Don't try to parse, convert or check - just do what you need to and add an except block to handle failures